taichi/codegen/spirv/kernel_utils.cpp

```diff
@@ -91,13 +97,47 @@ KernelContextAttributes::KernelContextAttributes(const Kernel &kernel)
     return bytes - offset;
   };
 
+  auto arange_rets = [](auto *vec, size_t offset) -> size_t {
+    std::vector<int> scalar_indices;
+    std::vector<int> array_indices;
+    for (int i = 0; i < vec->size(); ++i) {
```

Comment:
Do we need to put arrays after scalars? `arange_args` doesn't do so here.

Reply:
If we don't need that we can simplify the code in Metal backend as well.

Reply:
fix